### ISSUE TITLE
WIS2 Node broker links exposure in WCMP2 via GB subscriptions

### ISSUE THREAD

**Issue by tomkralidis** [wmo-im/wis2-guide#53]
cc @antje-s

Currently, when WCMP2 metadata is published, the workflow is that the GDC transposes/injects GB URLs for WCMP2 links (`origin` -> `cache`) which are bound to WIS2/MQTT/WTH subscription/download.  A link is added foreach GB that is part of the WIS2 global services ecosystem.  The "origin" link is never part of the resulting WCMP2 record in the GDC.

There is an edge case that one subscribing to the GB with topic `.../metadata/...`, will receive the "raw" WCMP2 document, which has the "origin" links.

A WIS2 Node may or may not want their MQTT endpoint made available.

In summary, the GDC updates the WCMP2 record links per above, but the GB does not.  Should it?

@6a6d74 @golfvert @kaiwirt

**Comment by golfvert** [comment]
GB should never modify messages. So, rewriting messages for metadata is a no-go for me.
We said that for all data, even without caching the data, GC would publish the message to `cache/a/wis2/...`
It means that access to topic `origin/a/wis2/...` is not needed for users. So, we could prevent access to the TH `origin/a/wis2/... `to the user everyone/everyone. And create a different user/pwd for those (GC, GDC) needing access to `origin/a/wis2/... `on the GBs. It would also avoid making the origin links available to users... Two birds with one stone ?

**Comment by tomkralidis** [comment]
So then, from a GB viewpoint `origin/a/wis2` will become protected for access by only WIS2 Global services?

Do we then need to introduce MQTT `origin/a/wis2` access control as a requirement for WIS2 Node brokers?  If WIS2 global services "guarantee" that clients will never find out about WIS2 Node brokers from the Global Services themselves, is this enough (for those who care)?

**Comment by antje-s** [comment]
GC will not modify data/metadata content, so that the content of metadata record files remains unchanged (including any existing links to origin WIS2 Node Message Brokers). On GCs those metadata files will be cached when origin notification is received and processed before the notification of the GDC (including the link to modified metadata record files). 
One possible solution from our last TT-WISMD meeting would be if the origin notifications contain properties.cache=false and the GDC notification not.
Another solution would be the proposed access limitation of the WIS2 Node Message Brokers to the GB addresses. However, the second solution would have the disadvantage that we may have some metadata in the GCs containing links that are not executable for consumers.

**Comment by 6a6d74** [comment]
Metadata records from `origin` will only be in the GC for 24-hours. We don't recommend that people subscribe to `origin/..../metadata` - the official place to get metadata is the GDC. Note that the GDC publishes a complete metadata set once per day - these records will include the injected links.

Closing this issue.
